### DISCUSSION TITLE
Handle existing user has no idp

### DISCUSSION THREAD

**Discussion by quangtruongnb** [sventorben/keycloak-home-idp-discovery#466] [General]
When the user enters an existing email and it has no idp, UserModel must be set to context for password form. And if email does not exist and email domain is not configured i think the "user doesn't exist message" should be displayed.

![image](https://github.com/user-attachments/assets/5becfb50-6b57-47b8-9af3-56cc3656204b)
![image](https://github.com/user-attachments/assets/db07d4dd-9836-45ab-9605-7d467104f938)

**Comment by sventorben** [comment]
@quangtruongnb can you please describe the overall behavior you would like to achieve with that? What are the requirements to request such changes?
If you just want username&password form as a fallback, you can just add it as an alternative authenticator.

**Comment by quangtruongnb** [comment]
@sventorben for normal user, hidp will redirect to their corresponding idp or use user/password form as expected but i think some cases can improve UX:

|No.| Input   |      account already exists    |  has corresponding domain  |  Expect | Actual |
| --- |-----------|:---------------------------------------:|-----------------------------------------:|-------------:|-------------:|
| 1 | username1@d1.com | yes  | no | user enters their password and can't change username using password form | Error: Invalid username or password. |
| 2 | username1@d2.com | no  | no | show error message "user doesn't exist message" | show user/password form|

**Reply by sventorben** [reply]
Could you post your flow configuration please? I think part of the user experience here is how you setup alternative autheticators.
